### DISCUSSION TITLE
Can't get summary from model even after it worked

### DISCUSSION THREAD

**Discussion by Hellobamboobamboo** [lnccbrown/HSSM#543] [General]
I ran the model, and got the model object, it showed the 4 options after, but still can't get summary. Also after a while, the model object showed not defined anymore. Why is that?
![Screenshot (16)](https://github.com/user-attachments/assets/c6739a11-9e1f-459a-ba17-0e6edf76926c)
![Screenshot (17)](https://github.com/user-attachments/assets/5ca628fd-495a-4f01-9e71-ab4e09bb6a7e)
![Screenshot (18)](https://github.com/user-attachments/assets/563c0ecf-762c-4e9c-b298-dd285e380004)
![Screenshot (19)](https://github.com/user-attachments/assets/80440c24-58a6-4d18-8624-7e19812f58cf)

**Comment by digicosmos86** [comment]
If you look at the index of the notebook cells, you will see that the cells were not run sequentially. Please restart the kernel and run all cells from the beginning of the notebook

**Comment by Hellobamboobamboo** [comment]
I did run the whole thing then still found that the summary doesn't work.
What could the reason be?

Sincerely,
Jingzhu Chen
(she/her/hers)
***@***.***
917 292 3896

M.S. in Applied Statistics for Social Science Research
NYU Fall 2023 Student Employee Excellence Award Winner
Former piano teacher of Sonoma School District
Former concert pianist of SF Bay Area
https://jingzhuchen.quarto.pub/personal-website/

**Reply by digicosmos86** [reply]
Can you share the screenshots of your sequential run? The error messages would be very helpful

**Comment by Hellobamboobamboo** [comment]
There's no error message. Usually it just shows nothing. And the long
message after taking the samples with all the chains just looks normal. If
you run the model object right after sampling, it shows the 4 categories
that you can click, but just when you run the za.summary(), it shows
nothing, not even an error message. And when you use .sample(), it said it
can't be used here for the model object. Is it a bug?



Sincerely,
Jingzhu Chen
(she/her/hers)
***@***.***
917 292 3896

M.S. in Applied Statistics for Social Science Research
NYU Fall 2023 Student Employee Excellence Award Winner
Former piano teacher of Sonoma School District
Former concert pianist of SF Bay Area
https://jingzhuchen.quarto.pub/personal-website/

**Comment by Hellobamboobamboo** [comment]
Also, the model object doesn't get saved after a few minutes. It used to be
able to be saved and you don't have to run the chain over again. Just
recently, just a few minutes after, when you run the model object, it says
the object is not defined.

Sincerely,
Jingzhu Chen
(she/her/hers)
***@***.***
917 292 3896

M.S. in Applied Statistics for Social Science Research
NYU Fall 2023 Student Employee Excellence Award Winner
Former piano teacher of Sonoma School District
Former concert pianist of SF Bay Area
https://jingzhuchen.quarto.pub/personal-website/

**Comment by digicosmos86** [comment]
![356816758-563c0ecf-762c-4e9c-b298-dd285e380004](https://github.com/user-attachments/assets/7b44df6e-1606-4009-8813-5d0210ce68a9)

From your screenshot, it seems that after sampling the model, the Python kernel was either restarted, or the notebook session was closed and started again, so the cell index is 2, which indicates that all the results from the previous cells were lost. The rest of the cells weren't even run after the cell indexed 2 threw an error.

Can you go to the `kernel` menu and `restart kernel and run all cells`?

**Comment by Hellobamboobamboo** [comment]
No, it's not that. I think it might be a bug. Everytime, every time after
the sampling is done, the infer_basic_model only worked once and showed the
inner list. If you click it again, it doesn't show anything anymore. Also
there's no error message for running the rest of the code from this point.
It happens every time I run all the cells for the past week. Please help
instruct what to do from this point on.



[image: Screenshot (19).png]




Sincerely,
Jingzhu Chen
(she/her/hers)
***@***.***
917 292 3896

M.S. in Applied Statistics for Social Science Research
NYU Fall 2023 Student Employee Excellence Award Winner
Former piano teacher of Sonoma School District
Former concert pianist of SF Bay Area
https://jingzhuchen.quarto.pub/personal-website/

**Comment by digicosmos86** [comment]
Okay. It's possible the computer could run out of memory and the kernel could get killed after doing the sampling. How about we do this, after sampling, simply do `basic_model.summary()` (without any parameters) instead of `az.summary`?

Another question: how much memory does the computer running the model have?
